internal/credential/vault/query.go

```diff
@@ -88,4 +88,34 @@ select *
   from credential_vault_library_private
  where public_id in (%s);
 `
+
+	updateTokenExpirationQuery = `
+update credential_vault_token 
+  set last_renewal_time = now(),
+  expiration_time = wt_add_seconds_to_now(?)
+  where token_hmac = ?;
+`
+
+	updateTokenStatusQuery = `
+update credential_vault_token 
+	set status = ? 
+	where token_hmac = ?;
+`
+
+	tokenRenewalNextRunInQuery = `
+with 
+	renewable_tokens(expiration_time, renewal_time) as (
+        select 
+		  expiration_time,
+		  last_renewal_time + (expiration_time - last_renewal_time) / 2 as renewal_time
+		from credential_vault_token
+		where status in ('current', 'maintaining')
+	)
+	select extract(epoch from renewal_time - now())::int as renewal_in
+  	  from renewable_tokens
+     where expiration_time = (
+       select min(expiration_time)
+         from renewable_tokens
+     );
```

Comment:
Can this `where` clause be moved to the `with renewable_tokens` part of the CTE?